lib/curly/parser.rb

```diff
@@ -136,6 +136,22 @@ def parse_inverse_conditional_block_start(*args)
     parse_block(:inverse_conditional, *args)
   end
 
+  def parse_else_block_start(*args)
+    block = @stack.pop
+
+    if block.nil?
+      raise Curly::Error, "An else needs to be in a block"
+    end
+
+    unless block.type == :conditional
```

Comment:
How about 'unless block && block.type == :conditional` then the first check can be removed.